src/libp2p/Libp2p.Protocols.TLS/TLS.cs

```diff
@@ -0,0 +1,408 @@
+        _logger?.LogDebug("TlsProtocol instantiated");
+        _sessionKey = ECDsa.Create();
+    }
+
+    public string Id => "/tls/1.0.0";
+
+
+
+    public async Task ListenAsync(IChannel downChannel, IChannelFactory? channelFactory, IPeerContext context)
+    {
+        _logger?.LogDebug("Handling connection");
+        _logger?.LogTrace("Successfully created client authentication options.");
+        if (channelFactory is null)
+        {
+            throw new ArgumentException("Protocol is not properly instantiated");
+        }
+
+        _logger?.LogDebug("Successfully received data from signaling channel.");
+        _logger?.LogDebug($"Successfully received data from signaling channel. Remote peer address: {context.RemotePeer.Address}");
+
+        Stream str = new ChannelStream(downChannel, _logger);
+
+        string remotePeerId = await downChannel.ReadLineAsync();
+        string remoteAddressString = context.RemotePeer.Address.ToString();
+        string fullAddressString = $"{remoteAddressString}/p2p/{remotePeerId}";
+
+
+        Multiaddress newAddressWithPeerId = Multiaddress.Decode(fullAddressString);
+
+
+        _logger?.LogDebug($"New address with PeerID: {newAddressWithPeerId}");
+        context.RemotePeer.Address = newAddressWithPeerId;
+
+
+        _logger?.LogDebug($"Successfully received data from signaling channel. Remote peer address: {remotePeerId}");
+
+        X509Certificate certificate = CertificateHelper.CertificateFromIdentity(_sessionKey, context.LocalPeer.Identity);
+
+        _logger?.LogDebug($"certificate: {certificate}");
+
+        SslServerAuthenticationOptions serverAuthenticationOptions = new()
+        {
+            ApplicationProtocols = [new SslApplicationProtocol("/yamux/1.0.0")],
+            RemoteCertificateValidationCallback = (_, certificate, _, _) => VerifyRemoteCertificate(context.RemotePeer.Address, certificate),
+            ServerCertificate = certificate,
+            ClientCertificateRequired = true,
+        };
+        _logger?.LogTrace("Successfully created client authentication options.");
+        // _logger?.LogDebug("Accepted new TCP client connection from {RemoteEndPoint}", tcpClient.Client.RemoteEndPoint);
+        SslStream sslStream = new(str, false, serverAuthenticationOptions.RemoteCertificateValidationCallback);
+        _logger?.LogTrace("Sslstream initialized.");
+        try
+        {
+            _logger?.LogTrace("Sslstream initialized.1");
+            await sslStream.AuthenticateAsServerAsync(serverAuthenticationOptions);
+            _logger?.LogTrace("Sslstream initialized.2");
+        }
+        catch (Exception ex)
+        {
+            _logger?.LogError("An error occurred during TLS authentication: {Message}", ex.Message);
+            _logger?.LogDebug("Exception details: {StackTrace}", ex.StackTrace);
+            throw;
+        }
+
+        if (sslStream.NegotiatedApplicationProtocol == SslApplicationProtocol.Http2)
```

Comment:
It's not expected to have http or any such kind of protocol here, it's for muxer. True for stuff below too